tasks/format_code.py

```diff
@@ -51,7 +51,12 @@ def format(ctx, check=False):
         .split("\n")[:-1]
     )
 
-    clang_cmd = "clang-format-10 -i {}".format(" ".join(files_to_check))
+    if check:
+        clang_cmd = "clang-format-10 --dry-run --Werror {}".format(
+            " ".join(files_to_check)
+        )
+    else:
+        clang_cmd = "clang-format-10 -i {}".format(" ".join(files_to_check))
```

Comment:
Could be cleaner as:

```python
clang_cmd = [
    "clang-format-10",
    "--dry-run --Werror" if check else "-i",
    files_to_check,
]
clang_cmd = " ".join(clang_cmd)
```

Is the `-i` not needed in the dry run command too though?

Reply:
Agree, your suggestion makes it cleaner. Afaict `-i` stands for "modify (files) in place" whereas `--dry-run` means "check for formatting changes, but don't apply any", so we don't need the `-i` in the `--dry-run`.